org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java

```diff
@@ -50,10 +53,22 @@ public List<Either<Command, CodeAction>> getCodeActions(ICodeActionService2.Opti
 		List<Either<Command, CodeAction>> actions = super.getCodeActions(options);
 		for (Diagnostic d : options.getCodeActionParams().getContext().getDiagnostics()) {
 			Object code = d.getCode().get();
-			if (TestLanguageValidator.INVALID_NAME.equals(code)) {
-				actions.add(Either.forLeft(fixInvalidName(d, options)));
-			} else if (TestLanguageValidator.UNSORTED_MEMBERS.equals(code)) {
-				actions.add(Either.forRight(fixUnsortedMembers(d, options)));
+			try {
```

Comment:
Can we leave this test-lang untouched to see whether the former way to write fixes still works?

Reply:
not quite because someone needs to create the Resource and the Document. I have submited a newer version where this is abstracted, but I still needed to change the signature of the method a little bit.